### PR TITLE
filetype: openCL files are not recognized

### PR DESCRIPTION
**Pull request by Freed-Wu**
Problem: OpenCL files are not recognized
Solution: recognize OpenCL files as opencl filetype

### CODE REVIEW

File: runtime/syntax/opencl.vim [hunk 1]
```diff
@@ -0,0 +1,11 @@
+" Vim syntax file
+" Language:	C
+
+if exists("b:current_syntax")
+  finish
+endif
+
+" TODO: support openCL specific keywords
+setlocal syntax=c
```
Comment (chrisbra) [inline runtime/syntax/opencl.vim]:
```suggestion
runtime! syntax/c.vim
```

File: runtime/syntax/opencl.vim [hunk 2]
```diff
@@ -0,0 +1,11 @@
+" Vim syntax file
+" Language:	C
```
Comment (chrisbra) [inline runtime/syntax/opencl.vim]: please add the usual Last Change and Maintainer header

File: runtime/ftplugin/opencl.vim [hunk 3]
```diff
@@ -0,0 +1,10 @@
+" Vim filetype plugin file
+" Language:	OpenCL
```
Comment (chrisbra) [inline runtime/ftplugin/opencl.vim:2]: please add the usual Last Change and Maintainer header

**Comment by chrisbra** [comment]
again, can you please add a test?

**Comment by chrisbra** [comment]
thanks, but please change the Maintainer line.

**Comment by zeertzjq** [comment]
The test doesn't work

**Comment by chrisbra** [comment]
thanks!

**Comment by chrisbra** [comment]
Actually, it seems the tests still don't work

File: src/testdir/test_filetype.vim [hunk 4]
```diff
@@ -1196,6 +1196,22 @@ func Test_cfg_file()
   filetype off
 endfunc
 
+func Test_cl_file()
+  filetype on
+
+  call writefile(['int f() {}'], 'Xfile.cl')
```
Comment (chrisbra) [inline src/testdir/test_filetype.vim]: that doesn't match the test in `FTcl()`

File: runtime/autoload/dist/ft.vim [hunk 5]
```diff
@@ -144,6 +144,14 @@ export def FTcfg()
   endif
 enddef
 
+export def FTcl()
+  if getline(1) ~= '/*'
```
Comment (chrisbra) [inline runtime/autoload/dist/ft.vim]:
```suggestion
  if getline(1) =~ '/\*'
```

File: runtime/autoload/dist/ft.vim [hunk 6]
```diff
@@ -144,6 +144,14 @@ export def FTcfg()
   endif
 enddef
 
+export def FTcl()
+  if getline(1) =~ '\/*'
```
Comment (chrisbra) [inline runtime/autoload/dist/ft.vim]:
```suggestion
  if getline(1) =~ '/\*'
```

**Comment by chrisbra** [comment]
Test still fails, probably because the `FTcl()` still uses the wrong pattern.

File: runtime/autoload/dist/ft.vim [hunk 7]
```diff
@@ -144,6 +144,14 @@ export def FTcfg()
   endif
 enddef
 
+export def FTcl()
+  if getline(1) =~ '/\*'
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim]: Is checking only the first line enough?

Comment (Freed-Wu) [reply runtime/autoload/dist/ft.vim]: the file shoule have a header:

```
/*
 * XXX blahblah
 */
```

It should only occur in first line?

**Comment by chrisbra** [comment]
thanks!

**Pull request closed without merging**